File: bestia-models/src/test/java/bestia/model/dao/PlayerBestiaDAOTest.java
package bestia.model.dao;

import java.util.Set;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import bestia.model.domain.Account;
import bestia.model.domain.PlayerBestia;
import bestia.model.domain.ConditionValues;

@RunWith(SpringRunner.class)
@SpringBootTest
@DataJpaTest
public class PlayerBestiaDAOTest {
	
	private final static String BESTIA_NAME = "test";
	private final static String BESTIA_UNKNOWN_NAME = "blablitest";

	@Autowired
	private PlayerBestiaDAO playerDao;
	
	@Before
	public void setup() {
		
		Account acc = new Account();
		acc.setEmail("test@test.net");
		
		PlayerBestia pb = new PlayerBestia();
		
		ConditionValues sv = new ConditionValues();
		sv.setCurrentHealth(10);
		sv.setCurrentMana(10);
			
		pb.setStatusValues(sv);
		pb.setLevel(10);
		pb.setName(BESTIA_NAME);
		
		acc.setMaster(pb);
		pb.setOwner(acc);
		
		playerDao.save(pb);
	}
	
	@Test
	public void findPlayerBestiasForAccount_unknownAcc_null() {
		final Set<PlayerBestia> bestias = playerDao.findPlayerBestiasForAccount(1337);
		Assert.assertTrue(bestias.size() == 0);
	}
	
	@Test
	public void findPlayerBestiasForAccount_knownAcc_bestias() {
		final Set<PlayerBestia> bestias = playerDao.findPlayerBestiasForAccount(1337);
		Assert.assertTrue(bestias.size() == 0);
	}
	
	@Test
	public void findMasterBestiaWithName_knownName_bestia() {
		PlayerBestia pb = playerDao.findMasterBestiaWithName(BESTIA_NAME);
		
		Assert.assertNotNull(pb);
	}
	
	@Test
	public void findMasterBestiaWithName_unknownName_null() {
		PlayerBestia pb = playerDao.findMasterBestiaWithName(BESTIA_UNKNOWN_NAME);
		Assert.assertNull(pb);
	}
}

File: bestia-models/src/test/kotlin/net/bestia/model/bestia/PlayerBestiaDAOTest.kt
package net.bestia.model.bestia

import net.bestia.model.IntegrationTest
import net.bestia.model.account.AccountRepository
import net.bestia.model.test.AccountFixture
import net.bestia.model.test.BestiaFixture
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired

@IntegrationTest
class PlayerBestiaDAOTest {

  @Autowired
  private lateinit var playerRepo: PlayerBestiaRepository

  @Autowired
  private lateinit var accountRepo: AccountRepository

  @Autowired
  private lateinit var bestiaRepository: BestiaRepository

  fun setupDatabase() {
    val acc = AccountFixture.createAccount(accountRepo)

    val pb = PlayerBestia(
        owner = acc,
        origin = BestiaFixture.createBestia(bestiaRepository)
    )

    val sv = ConditionValues(
        currentHealth = 10,
        currentMana = 10
    )

    pb.conditionValues = sv
    pb.level = 10

    playerRepo.save(pb)
  }

  @Test
  fun findPlayerBestiasForAccount_unknownAcc_null() {
    setupDatabase()
    val bestias = playerRepo.findAllByOwnerId(1337)
    assertTrue(bestias.isEmpty())
  }

  @Test
  fun findPlayerBestiasForAccount_knownAcc_bestias() {
    setupDatabase()
    val bestias = playerRepo.findAllByOwnerId(1337)
    assertTrue(bestias.size == 0)
  }
}
